src/index.test.js

```diff
@@ -0,0 +1,392 @@
+				'logging.GetLogHandlersReply',
+				'mesh.NetworkInfo',
+				'mesh.AuthRequest',
+				'mesh.AuthReply',
+				'mesh.CreateNetworkRequest',
+				'mesh.CreateNetworkReply',
+				'mesh.StartCommissionerRequest',
+				'mesh.StartCommissionerReply',
+				'mesh.StopCommissionerRequest',
+				'mesh.StopCommissionerReply',
+				'mesh.PrepareJoinerRequest',
+				'mesh.PrepareJoinerReply',
+				'mesh.AddJoinerRequest',
+				'mesh.AddJoinerReply',
+				'mesh.RemoveJoinerRequest',
+				'mesh.RemoveJoinerReply',
+				'mesh.JoinNetworkRequest',
+				'mesh.JoinNetworkReply',
+				'mesh.LeaveNetworkRequest',
+				'mesh.LeaveNetworkReply',
+				'mesh.GetNetworkInfoRequest',
+				'mesh.GetNetworkInfoReply',
+				'mesh.ScanNetworksRequest',
+				'mesh.ScanNetworksReply',
+				'mesh.DiagnosticType',
+				'mesh.DiagnosticInfo',
+				'mesh.GetNetworkDiagnosticsRequest',
+				'mesh.GetNetworkDiagnosticsReply',
+				'StorageType',
+				'SectionType',
+				'FirmwareModuleType',
+				'FirmwareModuleValidityFlag',
+				'SectionFlag',
+				'FileFormat',
+				'StartFirmwareUpdateRequest',
+				'StartFirmwareUpdateReply',
+				'FinishFirmwareUpdateRequest',
+				'FinishFirmwareUpdateReply',
+				'CancelFirmwareUpdateRequest',
+				'CancelFirmwareUpdateReply',
+				'FirmwareUpdateDataRequest',
+				'FirmwareUpdateDataReply',
+				'DescribeStorageRequest',
+				'DescribeStorageReply',
+				'ReadSectionDataRequest',
+				'ReadSectionDataReply',
+				'WriteSectionDataRequest',
+				'WriteSectionDataReply',
+				'ClearSectionDataRequest',
+				'ClearSectionDataReply',
+				'GetSectionDataSizeRequest',
+				'GetSectionDataSizeReply',
+				'GetModuleInfoRequest',
+				'GetModuleInfoReply'
+			]);
+		});
+	});
+
+	it('provides _pbjsJSON (parsed JSON rendered via `npm run build:json` command)', () => {
+		expect(DeviceOSProtobuf._pbjsJSON).to.be.an('object');
+	});
+
+	it('provides _pbjsObjects (generated Javascript via `npm run build` command)', () => {
+		expect(DeviceOSProtobuf._pbjsObjects).to.be.an('object');
+		expect(DeviceOSProtobuf._pbjsObjects).to.have.property('GetSerialNumberRequest');
+	});
+
+	describe('Usage of getDefinition() result', () => {
+		it('demonstrates how to use the request and replyMessage', () => {
+			const protobufDefinition = DeviceOSProtobuf.getDefinition('GetSerialNumberRequest');
+			assertValidProtobufMessage(protobufDefinition.message);
+			const msg = protobufDefinition.message.create({});
+			expect(msg).to.be.an.instanceOf(protobufDefinition.message);
+			const buffer = protobufDefinition.message.encode(msg).finish();
+			expect(buffer).to.be.an.instanceOf(Buffer);
+
+			// Nuance: We don't know how to stub the reply in these tests yet
+			// Normally, Device OS is responsible for creating the buffer representing
+			// the replyMsg.
+			// This code does not work because bufferFromDeviceOS is invalid
+			// It's unclear how to produce valid byutes to mock this for now
+			// const bufferFromDeviceOS = Buffer.alloc(15, '1');
```

Comment:
In a terminal, you can load the correct Response object and encode a response, store it in hex in this file like this: `Buffer.from('e73434490154', 'hex')`

Reply:
'preciate the comment. Though the solution you alluded to didn't work for me, I did manage to achieve what I was hoping for via https://github.com/particle-iot/device-os-protobuf/pull/10/commits/881d054face1833ccf6310dd995f877d605df1c9 👍